tree/dataframe/inc/ROOT/RDF/InternalUtils.hxx

```diff
@@ -0,0 +1,31 @@
+// Author: Vincenzo Eduardo Padulano CERN 02/2023
+
+/*************************************************************************
+ * Copyright (C) 1995-2023, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDF_INTERNALUTILS
+#define ROOT_RDF_INTERNALUTILS
+
+#include <utility>
+#include <ROOT/RDF/RInterface.hxx> // RNode
+
+namespace ROOT {
+namespace Internal {
+namespace RDF {
+
+/**
+ * \brief Change the range of empty entries that the dataframe will generate.
+ *
+ */
+void ChangeEmptyEntryRange(ROOT::RDF::RNode node, std::pair<ULong64_t, ULong64_t> &&newRange);
```

Comment:
RInterface.hxx already forward declares this function, what's the point of putting a copy into a dedicated header that includes RInterface.hxx? I might just not understand the design, could you explain?

Reply:
The new header is for the utilities that depend on `RInterface.hxx` (not the other way around like in the `InterfaceUtils.hxx` header. As for the forward declaration in `RInterface.hxx`, if I don't put that there then the friend declaration in the class will not match with any declaration. I guess your point is that I could just remove the `InternalUtils.hxx` header and have just `RDFInternalUtils.cxx` header that includes `RInterface.hxx` which will contain the declaration of this function?  In that case though I will have sort of an asymmetry between where the function is declared (i.e. inside RInterface.hxx) and where it is defined (in RDFInternalUtils.cxx) and I don't like that too much.

Another alternative is just declaring the function in RInterface.hxx and define it in RInterface.cxx, didn't think about that initially but I suppose it can be done in principle